examples/memcached.md

```diff
@@ -0,0 +1,197 @@
+# memcached
+
+As shown in [this blog
+post](https://memcached.org/blog/persistent-memory/), memcached can
+efficiently utilize PMEM as replacement for DRAM. This makes it
+possible to increase cache sizes and/or reduce costs. The memcached
+maintainers make container images available which support this
+feature. This example shows how to take advantage of PMEM with
+memcached running on Kubernetes.
+
+## PMEM as DRAM replacement
+
+Memcached can map PMEM into its address space and then store the bulk
+of the cache there. The advantage is higher capacity and (depending on
+pricing) less costs.
+
+The cache could be stored in a persistent volume and reused after a
+restart (see next section), but when that isn't the goal, a
+[deployment with one CSI ephemeral inline
+volume](../deploy/kustomize/memcached/ephemeral/memcached-ephemeral.yaml)
+per memcached pod is very simple:
+- A Kubernetes deployment manages the memcached instance(s).
+- In the pod template, one additional PMEM volume of a certain size is
+  requested.
+- An init container prepares that volume for use by memcached:
+  - It determines how large the actual file can be that memcached will use.
+  - It changes volume ownership so that memcached can run as non-root
+    process.
+- A wrapper script adds the `--memory-file` and `--memory-limit` parameters
+  when invoking memcached, which enables the use of PMEM.
+
+For this to work, the PMEM-CSI driver must have been deployed with
+`pmem-csi.intel.com` as driver name. No storage classes are
+needed. That name and other parameters in the deployment can be
+modified with
+[`kustomize`](https://github.com/kubernetes-sigs/kustomize). Here's
+how one can change the namespace, volume size or add additional
+command line parameters:
+
+```
+$ mkdir -p my-memcached-deployment
+$ cat >my-memcached-deployment/kustomization.yaml <<EOF
+namespace: demo
+bases:
+ - ../deploy/kustomize/memcached/ephemeral
```

Comment:
The implicit assumption here is that users check out the repo when following this example. Alternatively, once we have merged this we could use a `http://` URL that points into the repo, ideally with a specific commit hash.

In particular when reading this on our new docsite, a link into github *in the text* would be useful.

I think I'll try to add that separately.